### PR TITLE
implement SNDFONT lump

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: prboom2/src/MUSIC/flplayer.c [hunk 1]
```diff
@@ -105,9 +107,51 @@ static const char *fl_name (void)
 #include <delayimp.h>
 #endif
 
+static void *fl_sfopen(const char *lumpname)
```
Comment (kraflab) [inline prboom2/src/MUSIC/flplayer.c:110]: Let's pass lumpnum instead of lumpname here and save a lookup.

File: prboom2/src/memio.c [hunk 2]
```diff
@@ -198,7 +198,7 @@ int mem_fseek(MEMFILE *stream, signed long position, mem_rel_t whence)
       return -1;
   }
 
-  if (newpos < stream->buflen)
+  if (newpos <= stream->buflen)
```
Comment (kraflab) [inline prboom2/src/memio.c:201]: Could you explain this change / what consequences there might be?

File: prboom2/src/MUSIC/flplayer.c (hunk 1 above)
Comment (rfomin) [reply prboom2/src/MUSIC/flplayer.c:110]: This is a callback, so we need to pass `const char *`. We can save an int to a string, or convert pointers, or make `lumpnum` global, but I don't think it's worth it.

File: prboom2/src/memio.c (hunk 2 above)
Comment (rfomin) [reply prboom2/src/memio.c:201]: Without this change SEEK_END file will never work. Same fix in Odamex: https://github.com/odamex/odamex/blob/7190b0217abfd7fb22ad6f9f10026324367e8f3a/common/m_memio.cpp#L200

File: prboom2/src/MUSIC/flplayer.c (hunk 1 above)
Comment (kraflab) [reply prboom2/src/MUSIC/flplayer.c:110]: Ah ok, then all good 👍

**Review by kraflab — APPROVED** [review]
Thanks 🙌

**Pull request merged**